R/read_redcap.R

```diff
@@ -75,7 +79,8 @@ read_redcap <- function(redcap_uri,
                         export_survey_fields = NULL,
                         export_data_access_groups = NULL,
                         suppress_redcapr_messages = TRUE,
-                        guess_max = .Machine$integer.max) {
+                        guess_max = .Machine$integer.max,
+                        enable_mixed_structure = FALSE) {
```

Comment:
🏳️ Tried [`options`](https://dgkf.github.io/options/) package which seemed ok but I wasn't a huge fan of some of the outputs. Couldn't find a lot online for how to get these set internal to the package in just base R or many examples in some other larger packages. Feel free to send documentation my way if you know of any good ones.

Reply:
I was thinking it would be as simple as changing the default for `enable_mixed_structure` to be `getOption("redcaptidier.allow.mixed.structure", FALSE)` and then just adding to the docs that you can set the option with `options(redcaptidier.allow.mixed.structure = TRUE)`. Maybe that's not kosher for some reason though.

`options` package looks interesting but maybe not necessary if we're not managing a whole bunch of options.

Reply:
Got it, overcomplicated it as usual. I haven't found any examples of this so unsure if kosher or not, but check out the most recent commit and see what you think. In practice it seems to be fine, I don't know if there's anything around this we need to do for testing or whatnot. Made a brief updated to the `?read_redcap()` params docs but let me know if you think it needs something more robust or a callout in the vignette.